### ISSUE TITLE
Get rid of nginx on API

### ISSUE THREAD

**Issue by aapeliv** [Couchers-org/couchers#1344]
Currently requests go from client -> nginx -> envoy -> backend. That is, there's a double proxy.

After #1343, nginx is there just to terminate TLS (and handle media stuff, but it could still be there for media, that's a separate issue). This also includes automatic certificate renewal, etc. Envoy is more than capable of doing this without nginx.

This requires a bit of care; Envoy is not secure by default because it's used for a lot more than edge proxying, so someone needs to read up on how to secure it.
